pennylane_cirq/simulator_device.py

```diff
@@ -208,12 +211,60 @@ def generate_samples(self):
 
         self._result = self._simulator.run(self.circuit, repetitions=self.shots)
 
-        # Bring measurements to a more managable form, but keep True/False as values for now
-        # They will be changed in the measurement routines where the observable is available
         return np.array(
             [self._result.measurements[str(wire)].flatten() for wire in range(self.num_wires)]
         ).T.astype(int)
 
+    def expval(self, observable, shot_range=None, bin_size=None):
+        # pylint: disable=missing-function-docstring
+        # Analytic mode
+        if self.shots is None:
+            # Observables are in tensor form
+            if isinstance(observable, qml.operation.Tensor):
+                # Projector, Hamiltonian, Hermitian
+                for name in observable.name:
+                    if self._observable_map[name] is None or name == "Projector":
+                        return super().expval(observable, shot_range, bin_size)
+
+                if "Hadamard" in observable.name:
+                    T = qml.operation.Tensor()
+                    for obs in observable.obs:
+                        T = qml.operation.Tensor(T, qml.PauliZ(wires=obs.wires))
```

Comment:
What is the advantage to do it in this order?

Reply:
We're not calling on the `__init__` method of `qml.operation.Tensor` which might do some heavy lifting. At the moment it queues the constituent observables. Other than that, it would feel better to create a complex object as `qml.operation.Tensor` only once.